Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1175,6 +1157,93 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func sk2PromotionalOffer(forProductDiscount productDiscount: StoreProductDiscountType,
+                             discountIdentifier: String,
+                             product: StoreProductType,
+                             subscriptionGroupIdentifier: String,
+                             completion: @escaping @Sendable (Result<PromotionalOffer, PurchasesError>) -> Void) {
+        self.transactionFetcher.fetchLastVerifiedTransaction { transaction in
```

Comment:
We need at least one JWS token in the backend in order to sign the offer, plus we get to fail early if user has not made any purchase.